### ISSUE TITLE
Problems creating a new Elastic Beanstalk instance

### ISSUE THREAD

**Issue by indigobravo** [hypothesis/h-periodic#113]
Creating this issue for reference. The problem encountered is not a deal breaker, we have found a work around. More info can be found in slack: https://hypothes-is.slack.com/archives/CR3E3S7K8/p1625143504082300

**Overview**

A problem was encountered when trying to launch a new instance of h-periodic in Canada. It would appear that we have run into a catch22 situation.

Elastic Beanstalk expects the Docker container to run cleanly when starting up a new environment. However, it would appear h-periodic throws a hard exit code when it can not find dependant environment variables. The problem is you can not set EB environment variables until the environment has been created.

This is different behaviour from our other apps, via, h etc… When deploying those, the service just throws an error until the environment variables have been set.

Unfortunately, this is causing the EB environment to miss key initialization steps. Even setting the env var after the failure doesn’t fix the problem because EB didn't successfully configured the environment.

I am not sure of the solution here, but I am wondering if we can make h-periodic wait gracefully until the environment variables have been set?

**Comment by robertknight** [comment]
Ongoing discussion here: https://hypothes-is.slack.com/archives/CR3E3S7K8/p1625216559088600?thread_ts=1625143504.082300&cid=CR3E3S7K8

TL;DR - I think we should consider following this issue up with AWS before making changes to h-periodic. h-periodic's behavior of crashing on startup if required environment variables are not found seems quite reasonable to me. We should also make sure that we are using the latest EB platform version in case AWS have fixed the issue already.

**Comment by indigobravo** [comment]
> We should also make sure that we are using the latest EB platform version in case AWS have fixed the issue already.

This solved the problem. Upgrading to the latest image still results in a failure during the initial build of the environment. However, EB goes into a green / ok state where you can set the environment variables.

The error thrown has now changed to:
```
Instance deployment: The Docker container unexpectedly ended after it was started.
```
